src/main.rs

```diff
@@ -33,35 +227,56 @@ fn main() -> OrtResult<()> {
             ExecutionProvider::CoreML(Default::default()),
             ExecutionProvider::CPU(Default::default()),
         ])?
-        .with_model_from_file(onnx_model_file)?; // Load the ONNX model from a file
+        .with_model_from_file(onnx_model_file)?;
+    Ok(session)
+}
+
+// Function to find the bounding box containing non-transparent pixels
+fn find_alpha_bounds(image: &RgbaImage) -> Option<(u32, u32, u32, u32)> {
+    let mut min_x = u32::MAX;
+    let mut max_x = 0;
+    let mut min_y = u32::MAX;
+    let mut max_y = 0;
 
-    // Get the required input shape: [batch_size, channel, height, width]
+    for (x, y, pixel) in image.enumerate_pixels() {
+        if pixel[3] > 10 {
```

Comment:
How did you have this number? 10 is like hardcoded?

Reply:
Yes 10 was a random low number I chose and have tweaked but not gone to the level of parameterizing or adaptive. 

Do you have issue with MIT license?  I have a bunch of other stuff I could contribute but I'm hesitant with GPL.

Reply:
> Yes 10 was a random low number I chose and have tweaked but not gone to the level of parameterizing or adaptive.

To make it a CLI tool I think we need to find a way that can generalize well to most cases. I can help look into the technical report of the model or some other ways to find bounding boxes.

> Do you have issue with MIT license? I have a bunch of other stuff I could contribute but I'm hesitant with GPL.

No problem at all, it's now under MIT.

Reply:
This repo also uses a threshold of 10 https://github.com/danielgatis/rembg/blob/0d0f6c66f4a7bebd84bd40b7c958f084fdc02c9b/rembg/bg.py#L225

Reply:
I'm not sure the weights/models would force the code to use GPL.  https://github.com/xuebinqin/DIS is Apache 2, would that be a possibility?  What code came from the js projection?  It feels like history belongs to the DIS and less the javascript project.

I was able to get Cuda working, which takes some time to spin up (40s).  I added a http option that allows multipart-form image upload and responds with the image and the headers include the crop information (.5 seconds).  

I was able to change the geometry of the model.  I used "Colab_Demo.ipynb" and added the following code:
```
batch_size=1
x = torch.randn(batch_size, 3, 768, 768, requires_grad=False)
torch_out = net(x)

# Export the model
torch.onnx.export(net,               # model being run
                  x,                         # model input (or a tuple for multiple inputs)
                  "yourmodel.onnx",   # where to save the model (can be a file or file-like object)
                  export_params=True,        # store the trained parameter weights inside the model file
                  opset_version=11,          # the ONNX version to export the model to
                  do_constant_folding=True,  # whether to execute constant folding for optimization
                  input_names = ['input'])#,   # the model's input names```
```
The reshaping to 1024 is taking time, I played with lowering the dimensions to 768x768.
I would really like to learn how to train the model myself.  I haven't figured that out yet.

Reply:
> To make it a CLI tool I think we need to find a way that can generalize well to most cases

We can add a command line parameter?  I welcome any improvements to it, it works pretty good in my testing.

Reply:
> This repo also uses a threshold of 10 https://github.com/danielgatis/rembg/blob/0d0f6c66f4a7bebd84bd40b7c958f084fdc02c9b/rembg/bg.py#L225

Nice to do some erosion and uses of two foreground and background thresholds.  That would be cool to implement.

Reply:
> We can add a command line parameter? I welcome any improvements to it, it works pretty good in my testing.

Agree! could you please make a parameter and assign 10 as default value to the current PR?

Reply:
> I'm not sure the weights/models would force the code to use GPL. https://github.com/xuebinqin/DIS is Apache 2, would that be a possibility? What code came from the js projection? It feels like history belongs to the DIS and less the javascript project.

My concern is if they trained DIS using their own dataset then we need to follow their license or at least to have a discussion with them about license. I'm gonna ask them about this.

Reply:
Another solution is to switch to use DIS weights completely. But I haven't used DIS weight yet so not sure about its quality.